core/jobs/batch_jobs/rejecting_suggestion_for_invalid_content_ids_jobs.py

```diff
@@ -0,0 +1,515 @@
+                    utils.unescape_html(tag['content-with-value'])
+                )
+                collapsible_content = json.loads(
+                    collapsible_content_json)
+                if (
+                    RejectSuggestionWithMissingContentIdJob.
+                    _is_tag_removed_with_empty_content(
+                        tag, collapsible_content, is_collapsible=True)
+                ):
+                    continue
+
+                collapsible_content = (
+                    RejectSuggestionWithMissingContentIdJob.fix_rte_tags(
+                        collapsible_content,
+                        is_tags_nested_inside_tabs_or_collapsible=True)
+                )
+                if (
+                    RejectSuggestionWithMissingContentIdJob.
+                    _is_tag_removed_with_empty_content(
+                        tag, collapsible_content, is_collapsible=True)
+                ):
+                    continue
+
+                collapsible_content_json = json.dumps(collapsible_content)
+                tag['content-with-value'] = utils.escape_html(
+                    collapsible_content_json)
+            else:
+                tag.decompose()
+                continue
+
+            if (
+                RejectSuggestionWithMissingContentIdJob.
+                _is_tag_removed_with_invalid_attributes(
+                    tag, 'heading-with-value')
+            ):
+                continue
+
+        return str(soup).replace('<br/>', '<br>')
+
+    @staticmethod
+    def _fix_content(html: str) -> str:
+        """Helper function to fix the html.
+
+        Args:
+            html: str. The html data to fix.
+
+        Returns:
+            html: str. The fixed html data.
+        """
+        html = RejectSuggestionWithMissingContentIdJob.fix_rte_tags(
+            html, is_tags_nested_inside_tabs_or_collapsible=False)
+        html = (
+            RejectSuggestionWithMissingContentIdJob.
+            fix_tabs_and_collapsible_tags(html))
+        return html
+
+    @staticmethod
+    def _update_suggestion_model(
+        suggestions: List[suggestion_models.GeneralSuggestionModel],
+        exp_model: exp_models.ExplorationModel
+    ) -> None:
+        """Updates the translation suggestion. The translation whose
+        content_id no longer exists, the suggestion status will be marked
+        as `rejected`. The RTE content of the suggestion will be updated
+        in case invalid data is present.
+
+        Args:
+            suggestions: list(GeneralSuggestionModel). A list of translation
+                suggestion models corresponding to the given exploration.
+            exp_model: ExplorationModel. The exploration model.
+
+        Returns:
+            suggestions. List[GeneralSuggestionModel]. Result containing the
+            list of updated suggestion models.
+        """
+        exp_domain = exp_fetchers.get_exploration_from_model(exp_model)
```

Comment:
hmmm... this will migrate the exploration and will fix issues in exp, right? Do we want to do that?

Reply:
No, we cannot rely on v54 data for migration, is it possible to write this migration job to work with v52 of the state?